core/domain/classifier_domain.py

```diff
@@ -277,6 +277,27 @@ def status(self):
     def training_data(self):
         return self._training_data
 
+    def update_state_name(self, state_name):
```

Comment:
@anmolshkl , Ahh. Yeah, for most cases, I don't see where we would be updating state_name. I'm not sure of where it happens either. I added that function because I though we agreed that state_name was mutable. I don't see the mutator function doing much harm though.